### PR TITLE
CodeWhisperer: enhanced Java file fetching

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
- Config chunk size from 3000 to 60
- Enable Java CrossFile feature for 100% users
- Enable python CrossFile feature for 50% users
- Enable javascript CrossFile feature for 50% users
- Enable typescript CrossFile feature for 50% users

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileCrawler.kt [hunk 1]
```diff
@@ -62,6 +66,22 @@ abstract class CodeWhispererFileCrawler : FileCrawler {
         }
     }.orEmpty()
 
+    override fun listRelevantFilesInEditors(psiFile: PsiFile): List<VirtualFile> {
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileCrawler.kt:69]: Does this need a certain thread to execute in?

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileCrawler.kt:69]: good catch, let me wrap L72 in readAction, should be safer

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/TypescriptCodeWhispererFileCrawler.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.util
+
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+
+object TypescriptCodeWhispererFileCrawler : CodeWhispererFileCrawler() {
+    override val fileExtension: String = ".js"
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/TypescriptCodeWhispererFileCrawler.kt:10]: should update

**Review by andrewyuq — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt [hunk 3]
```diff
@@ -108,11 +114,39 @@ class DefaultCodeWhispererFileContextProvider(private val project: Project) : Fi
...
+                    extractSupplementalFileContextForSrc(psiFile, targetContext)
+                } else {
+                    emptyList()
+                }
+
+                is CodeWhispererTypeScript -> if (userGroup == CodeWhispererUserGroup.CrossFile) {
+                    extractSupplementalFileContextForSrc(psiFile, targetContext)
+                } else {
+                    emptyList()
+                }
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:146]: Can these be combined into one is block using 'or'

**Review by manodnyab — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt (hunk 3 above)
Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt:146]: sg

**Pull request merged**